tests/tt_metal/tt_metal/unit_tests_fast_dispatch/common/command_queue_fixture.hpp

```diff
@@ -109,3 +116,44 @@ class CommandQueuePCIDevicesFixture : public ::testing::Test {
     tt::ARCH arch_;
     size_t num_devices_;
 };
+
+class CommandQueueSingleCardFixture : public ::testing::Test {
+   protected:
+    void SetUp() override {
+        auto slow_dispatch = getenv("TT_METAL_SLOW_DISPATCH_MODE");
+        if (slow_dispatch) {
+            TT_THROW("This suite can only be run with fast dispatch or TT_METAL_SLOW_DISPATCH_MODE unset");
+            GTEST_SKIP();
+        }
+        arch_ = tt::get_arch_from_string(tt::test_utils::get_env_arch_name());
+
+        const chip_id_t mmio_device_id = 0;
+        std::vector<chip_id_t> device_ids_to_open = {mmio_device_id};
+
```

Comment:
I have an api to do this:
```
std::vector<chip_id_t> chip_ids;
        for (unsigned int id = 0; id < num_devices_; id++) {
            chip_ids.push_back(id);
        }
        reserved_devices_ = tt::tt_metal::detail::CreateDevices(chip_ids);
        for (const auto& id : chip_ids) {
            devices_.push_back(reserved_devices_.at(id));
        }
        // skip tests if no ethernet cores

    }

    void TearDown() override { tt::tt_metal::detail::CloseDevices(reserved_devices_); }

    std::vector<tt::tt_metal::Device*> devices_;
    std::map<chip_id_t, tt::tt_metal::Device*> reserved_devices_;
```

Reply:
This is done for `CommandQueuePCIDevicesFixture` which I hope to delete once we have support for remote

Reply:
`CreateDevices` will call `tt::Cluster::instance().set_internal_routing_info_for_ethernet_cores(true);` as well